[WIP] Improved work with ConnectionState

vladimirzenin opened the pull request:
Subscription to ConnectionState has been added.
Added display of the connection status in the bottom left corner.

x2bool commented on services/Tel.Egram.Utils/TdLib/IAgent.cs:
```diff
@@ -7,6 +7,8 @@ public interface IAgent
     {
         IObservable<TdApi.Update> Updates { get; }
 
+        IObservable<TdApi.ConnectionState> ObserveConnectionState();
```
Agent isn't meant to be used directly. It's purpose is to execute actions and receive updates. Instead the preferred approach is to create a wrapper around it. You could see how it's done with IAuthenticator for example.

x2bool commented on components/Tel.Egram.Components.Application/ApplicationController.cs:
```diff
@@ -142,6 +175,31 @@ private void GoToWorkspacePage()
 
+        private void UpdateConnectionState(int state)
+        {
+            if (Model.AuthenticationModel == null)
```
I don't understand why AuthenticationModel is getting activated by this method. Is your goal just to update TextBlock's value?

x2bool commented on components/Tel.Egram.Components.Application/ApplicationController.cs:
```diff
@@ -142,6 +175,31 @@ private void GoToWorkspacePage()
+
+            _workspaceController?.Dispose();
+
+            Model.StartupModel = null;
```
Why Workspace and Startup models are getting nulled by connection state changes? This causes white screen instead of chat content. I think updating ConnectionState should be enough.

x2bool commented on components/Tel.Egram.Components.Application/ApplicationController.cs:
```diff
@@ -28,18 +29,50 @@ public class ApplicationController : BaseController<MainWindowModel>
+                    switch (state)
+                    {
+                        case TdApi.ConnectionState.ConnectionStateConnecting _:
+                            UpdateConnectionState(0);
```
Personally I would create an enum representing connection state instead of using numbers.

vladimirzenin commented:
1. Fixed error with activated and getting nulled by AuthenticationModel.
2. Added enum ConnectionState.
3. Deleted changed in Agent and IAgent; Implemented subscribe to UpdateConnectionState in ApplicationController.
